core/client/hdfs/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java

```diff
@@ -585,6 +587,46 @@ public void getBlockLocationsNoUfsLocationsWithFallback() throws Exception {
     }
   }
 
+  @Test
+  public void appendExistingNotSupported() throws Exception {
+    Path path = new Path("/file");
+    alluxio.client.file.FileSystem alluxioFs =
+        mock(alluxio.client.file.FileSystem.class);
+    when(alluxioFs.exists(new AlluxioURI(HadoopUtils.getPathWithoutScheme(path))))
+        .thenReturn(true);
+    FileSystem alluxioHadoopFs = new FileSystem(alluxioFs);
+
+    try {
+      alluxioHadoopFs.append(path, 100);
+      fail("append() of existing file is expected to fail");
+    } catch (IOException e) {
+      assertEquals("append() to existing Alluxio path is currently not supported: " + path,
+          e.getMessage());
+    }
+    alluxioHadoopFs.close();
+  }
+
+  @Test
+  public void createWithoutOverwrite() throws Exception {
+    Path path = new Path("/file");
+    alluxio.client.file.FileSystem alluxioFs =
+        mock(alluxio.client.file.FileSystem.class);
+    when(alluxioFs.exists(new AlluxioURI(HadoopUtils.getPathWithoutScheme(path))))
+        .thenReturn(true);
+    when(alluxioFs.createFile(eq(new AlluxioURI(HadoopUtils.getPathWithoutScheme(path))), any()))
+        .thenThrow(new FileAlreadyExistsException(path.toString()));
+    FileSystem alluxioHadoopFs = new FileSystem(alluxioFs);
+
+    try {
```

Comment:
Shall we use junit `expectedMessage`?
```
expectedEx.expectMessage("message to expect");
```
see
https://stackoverflow.com/questions/2469911/how-do-i-assert-my-exception-message-with-junit-test-annotation

Reply:
In general, I have been favoring the manual catching, mainly because the `ExpectedException` only works with 1 exception per test. It is common that tests might expect multiple exceptions in a test, so I tend to manually catch the Exception. In these specific tests, we only deal with 1 Exception, so `ExpectedException` would have worked, but I usually avoid using it because of that limitation.